src/test/java/uk/nhs/hee/tis/trainee/forms/api/FormRPartAResourceTest.java

```diff
@@ -159,15 +154,7 @@ void postShouldNotCreateFormWhenFormExists() throws Exception {
 
   @Test
   void postShouldNotCreateFormWhenDtoValidationFails() throws Exception {
-    dto.setId(null);
-
-    BeanPropertyBindingResult bindingResult = new BeanPropertyBindingResult(dto, "formDto");
-    bindingResult.addError(new FieldError("formDto", "formField", "Form field not valid."));
-
-    Method method = validator.getClass().getMethod("validate", FormRPartADto.class);
-    Exception exception = new MethodArgumentNotValidException(new MethodParameter(method, 0),
-        bindingResult);
-    doThrow(exception).when(validator).validate(dto);
+    when(service.save(dto)).thenThrow(MongoWriteException.class);
```

Comment:
I don't quite get the verification here... we are calling `verifyNoInteractions(service)` which I would have thought means `service.save()` is never invoked.  Given the `postShouldCreate...` succeed, I can't see it being an obvious cause.

Perhaps it's worth adding an `Expect` for the message?

Reply:
Ah, I've seen that the test is doing something a bit different now... it is using the validation in the Resource.  That makes this line redundant.

Reply:
Good spot, its ended up testing the wrong thing after my change and was passing because of POSTing with an ID set rather than passing because the Mongo exception was well handled (there's a case for monitoring branch coverage instead of line coverage in there 😁 ).

I've added the `null` ID back in and changed the verifications to reflect what we actually expect to happen.